scripts/core/editor3/directive.tsx

```diff
@@ -204,7 +204,9 @@ class Editor3Directive {
             const props = {
                 item: this.item,
                 pathToValue: this.pathToValue,
+                value: $scope.vm.value,
```

Comment:
to make it read the value from item where it's updated and not meta fields where it's empty (after reset)

Reply:
Are there multiple sources for the same data? Isn't `pathToValue` meant as a guide to where to read the data from? Would it be possible then remove `pathToValue` if you are passing the value directly?

Reply:
there are times when it can be used and fields meta has correct value, but not after editing in that modal

Reply:
Which question of the 3 I asked are you answering? :smile:

I assume you agree that there are multiple objects holding the same data. Could it be reduced to one? If you found that the data source that was used before contains incorrect data it would be best to either fix the original one or switch to reading from another object completely.

Reply:
yes there are multiple objects with field data - the item itself and then there is fields_meta with draftjs editor state - which gets out of sync here. not sure we can get away without any of them.

Reply:
I finally understand what you are referring to, that is that we have draftjs data and HTML which is generated from it. We need draftjs data because comments/annotations/suggestions aren't in the HTML output, and we need HTML, because we want to be compatible with editor2 output.

What about if you removed draftjs data and re-initialized the draftjs field? If it can't find draftjs data it will read from HTML by default even without adding `value: $scope.vm.value`, won't it?

Reply:
that's exactly what I'm doing, but it didn't read it without adding that value,
that's why I've added it.. :wink:

Reply:
I'd like to prevent adding `value: $scope.vm.value,` if possible since it can affect all editor3 fields in unexpected ways. Could you debug to see why can't it load from HTML without this additional line?

Reply:
it would only read it from there https://github.com/superdesk/superdesk-client-core/blob/31a2fd9046e0b49c7e5c3b55bebac36178be8d8c/scripts/core/editor3/store/index.ts#L262-L265

Reply:
`value` wasn't in the props before. How did it read it then? Would it be possible to make a change to remove `value` property completely and read from `item[pathToValue]` instead? If not, add a comment next to `value: $scope.vm.value,` describing the use case you are adding it for.